### PR TITLE
Maria dev

### PR DESCRIPTION
**Pull request by Maria-Chira**
RegisterForm component

## Description
- added **validation** for the register form;
- created **server route** for the register form;
- updated data.sql file:  created **queries** to populate the **cohort table**; 
- 2 packages installed: **cors** and **bcrypt**;
- **hashed** the password.

## Related to
Ticket: RegisterForm Server

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

**Review by Johng117 — APPROVED** [review]
Really good job Maria, really logical and I like the password hash.

**Review by TomRafferty — APPROVED** [review]
Brilliant work Maria! This took a serious amount of effort 😮

File: server/api.js [hunk 1]
```diff
@@ -12,14 +14,61 @@ router.get("/events", (req,res)=> {
...
+		.query(getCohortId, [classNr, region])
+		.then((result) => result.rows[0].id);
+
+	//check if email is already used
+	pool
+		.query("SELECT * FROM users WHERE user_email = $1", [email])
+		.then((result) => {
+			if (result.rows.length > 0) {
+				return res
+					.status(400)
```
Comment (milleymilli) [inline server/api.js:52]: In this case status code must be 409, there is a conflict coz there is already existed email but 400 is bad request (example due to a bad syntax...etc)

**Review by milleymilli — CHANGES_REQUESTED** [review]
### **First Of All**
 Create one folder for all the routes 
- inside there create a folder for each tables 
1.  user_lists
2. events and so on 
and in each tables create their own routes 
**This way you will have everything organised  **
### **After Doing This You Need To Create Your CustomMiddleWare File  **
-  In this file you create your error handlers 
- and by doing this when ever you catch an error in your api you catch you direct it here. 

### _**REASONS FOR THIS IN GENERAL IS TO LIVE THE  API FILE TAKES CONTROL ONLY FOR THE ROOT URI ONLY AND IT WILL BE EASY TO DEBUG AND TO NAVIGATE  **__

File: client/src/components/RegisterForm.js [hunk 2]
```diff
@@ -10,125 +14,232 @@ const roles = ["Admin", "Student"];
...
+	const isRetypedPasswordValid = (retypedPassword) =>
+		formData.password == retypedPassword && retypedPassword.length > 5
+			? true
+			: false;
+
+
+	//calling all the validation functions in order to set the states for errors
+	//returning a boolean so I can do a general validation when the user try to submit the form
+	const areTheFieldsValid = () => {
+		isNameValid(formData.name) ? setNameError(false) : setNameError(true);
```
[the remark this answers is not in the document]
Comment (Maria-Chira) [reply client/src/components/RegisterForm.js:75]: Thanks for this. I'll leave this as a stretch goal if we will still have enough time.

File: server/api.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Maria-Chira) [reply server/api.js:52]: I'll make sure I'll include this changes in my next PR

Comment (Maria-Chira) [reply server/api.js:52]: Same here

**Pull request merged**